Document objects with memory

ThomasBreuer opened the pull request:
addresses #6173

ThomasBreuer committed:
document objects with memory

fingolfin commented on lib/memory.gi:
```diff
@@ -220,11 +217,12 @@ InstallMethod( InverseOp, "objects with memory", true,
                             InverseOp( a!.el ) );
     else
       return ObjWithMemory( slp, 0, a!.el );
+#T why can't we return a?
```
```suggestion
      return a;
```

ThomasBreuer commented on lib/memory.gd:
```diff
@@ -13,20 +13,261 @@
+
+DeclareOperation( "ForgetMemory", [ IsObject ] );
+
+DeclareGlobalFunction( "StripStabChain" );
```
Logically, `StripStabChain` does what `ForgetMemory` is intended for.
However, stabilizer chains are plain records, and installing a method for a record would require some additional checks. Since `StripStabChain` is not used in the library, I left the name and kept the function undocumented.

ThomasBreuer committed:
address comments

Concerning the question what to do when objects with memory
cause GAP errors,
add and document `MethodsForObjWithMemory`.

For better reference, document also `MethodsOperation`.

And fix some GAPDoc syntax.

ThomasBreuer committed:
changed a reference
